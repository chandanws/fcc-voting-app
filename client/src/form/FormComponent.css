.form {
  display: flex;
  flex-direction: column;
  border: 1px solid black;
  padding: 1rem;
}

.form__wrapper {
  margin: 1rem 0;
  display: flex;
  flex-direction: column;
}

.form__label {
  font-size: 1.5rem;
  margin-bottom: 1rem;
}

.form__input {
  font-size: 2rem;
  border: 1px solid black;
  padding: 1rem;
}

.form__submit {
  margin-top: 1rem;
  font-size: 2rem;
  background-color: lightgrey;
  border: 1px solid black;
  padding: 1rem;
}
