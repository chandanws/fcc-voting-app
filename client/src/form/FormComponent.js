import React from "react";
import PropTypes from "prop-types";
import "./FormComponent.css";
import { useModifierWithBlock } from "../helpers/helpers";

export const Form = ({ children, disabled, className, ...props }) => {
  const submit = disabled ? (
    <input disabled className="form__submit" type="submit" value="submit" />
  ) : (
    <input className="form__submit" type="submit" value="submit" />
  );
  return (
    <form className={`form ${className}`} {...props}>
      {children}
      {submit}
    </form>
  );
};

Form.propTypes = {
  onSubmit: PropTypes.func.isRequired
};

export const FormElement = ({
  component: Component,
  label,
  type,
  onChange,
  value,
  id,
  validation,
  children
}) => {
  let errorClassName;
  if (value === "" || validation === "") {
    errorClassName = "";
  } else if (validation === true) {
    errorClassName = "form__input--success";
  } else {
    errorClassName = "form__input--error";
  }

  return (
    <div className="form__wrapper">
      <label className="form__label">{label}</label>
      <div className="input-wrapper">
        <Component
          className={`form__input ${errorClassName}`}
          type={type}
          value={value}
          onChange={onChange(id)}
        />
        {children}
      </div>
    </div>
  );
};

FormElement.propTypes = {
  label: PropTypes.string.isRequired,
  type: PropTypes.string
};

FormElement.defaultProps = {
  type: "text",
  component: "input",
  validation: ""
};
