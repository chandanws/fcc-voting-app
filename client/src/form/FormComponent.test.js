import React from "react";
import { Form, FormElement } from "./FormComponent";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe("<Form />", () => {
  it("should render correctly", () => {
    const mockCallback = jest.fn();
    const wrapper = shallow(<Form onSubmit={mockCallback} />);
    expect(wrapper).toMatchSnapshot();
  });
});

// describe("<FormElement />", () => {});
