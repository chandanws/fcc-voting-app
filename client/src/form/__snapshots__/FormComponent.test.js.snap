// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> should render correctly 1`] = `
<form
  className="form undefined"
  onSubmit={[Function]}
>
  <input
    className="form__submit"
    type="submit"
    value="submit"
  />
</form>
`;
